docs/styleguide-cpp.md

```diff
@@ -1,320 +1,660 @@
+contributors, to minimize ambiguity, and to have a consistant way to
+convey the intent of code.
+
+## 1. Conventions
 
-	1.2 [File Formatting](styleguide-cpp.md#12-file-formatting)
+### 1.1. General Principles
 
- 2. [Conventions](styleguide-cpp.md#2-conventions)
+1.  Prefer clarity over brevity
 
-	2.1. [Naming Conventions](styleguide-cpp.md#21-naming-conventions)
+2.  [Don’t optimize prematurely](https://youtu.be/tKbV6BpH-C8)
 
-	2.2 [Do's and Don'ts](styleguide-cpp.md#22-dos-and-donts)
+3.  In cases of ambiguity, use your best judgement
 
-## 1. Why Style?
->“Programs are meant to be read by humans and only incidentally for computers to execute.”
-> - H. Abelson and G. Sussman in [*Structure and Interpretation of Computer Programs*](https://en.wikipedia.org/wiki/Structure_and_Interpretation_of_Computer_Programs)
+### 1.2. File Formatting
 
-The purpose of this styleguide is to give a clear framework for other contributors, to minimize ambiguity, and to have a consistant way to convey the intent of code.
+Source code files should adhere to the following:
 
-### 1.1. General Principles
-- Prefer clarity over brevity
-- [Don’t optimize prematurely](https://youtu.be/tKbV6BpH-C8)
-- In cases of ambiguity, use your best judgement
+1.  Encoded in UTF-8
 
-### 1.2. File Formatting
-Source code files should adhere to the following:
+2.  Use tabs for indentation, only use one tab per indent level
+
+3.  Use LF for end-of-line sequences
+
+4.  No trailing whitespace (Lines which end in spaces or tabs)
+
+5.  Any `#include` directives should be at the top of the file
+
+6.  Any header files have the `.hpp` file extension
+
+7.  Any implementation files have the `.cpp` file extension
+
+8.  Code should be within the `OpenVic2` namespace (or a nested
+    namespace)
+
+9.  Any submodules or third-party utilities should be in the
+    `OpenVic2/extension/deps` directory
+
+10. Groups of related source files should be kept in a sub-directory of
+    `OpenVic2/extension/src`
+
+11. Any lines that are longer than 120 characters in length should be
+    split across multiple lines
+
+12. Use British English spellings and standards.
+
+### 1.3. Naming Conventions
+
+|        |                                   |                                     |                                         |
+|--------|-----------------------------------|-------------------------------------|-----------------------------------------|
+| Number | Item                              | Writing Convention                  | Example                                 |
+| 1      | Class names                       | PascalCase                          | MyCoolExample                           |
+| 2      | Struct names                      | PascalCase                          | MyCoolExample                           |
+| 3      | Variable names                    | Lower snake_case                    | my_cool_example                         |
```

Comment:
Changing those conventions is totally subjective preference, I would stick with what @ClarkeCode decided on

Reply:
Consistent standards beg to differ.

Reply:
> Consistency standards beg to differ.

To what consistency standards are you referring?
If you are talking about keeping C++ code consistent with GDscript, there is no extrinsic or functional reason to do so.

This is subjective preference.

Reply:
And C++ code with other C++ code, it is useless to interface with Godot and violate what it does as it only creates confusion and requires nearly doubly the documentation work and at least twice as much work dealing with error messages.

Reply:
I personally find lower snake case easier to read.

Reply:
The C++ core guidelines (https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines) is directing people to underscore_style these days, and strongly discourages ALL_CAPS for enum constants. Just FYI